modules/pricer/src/test/java/com/opengamma/strata/pricer/rate/bond/DiscountingFixedCouponBondTradePricerTest.java

```diff
@@ -0,0 +1,323 @@
+  private static final CurveInterpolator INTERPOLATOR = Interpolator1DFactory.LINEAR_INSTANCE;
+  private static final CurveName NAME_REPO = CurveName.of("TestRepoCurve");
+  private static final CurveMetadata METADATA_REPO = Curves.zeroRates(NAME_REPO, ACT_365F);
+  private static final InterpolatedNodalCurve CURVE_REPO = InterpolatedNodalCurve.of(
+      METADATA_REPO, new double[] {0.1, 2.0, 10.0}, new double[] {0.05, 0.06, 0.09}, INTERPOLATOR);
+  private static final DiscountFactors DSC_FACTORS_REPO = ZeroRateDiscountFactors.of(EUR, VALUATION, CURVE_REPO);
+  private static final BondGroup GROUP_REPO = BondGroup.of("GOVT1 BOND1");
+  private static final CurveName NAME_ISSUER = CurveName.of("TestIssuerCurve");
+  private static final CurveMetadata METADATA_ISSUER = Curves.zeroRates(NAME_ISSUER, ACT_365F);
+  private static final InterpolatedNodalCurve CURVE_ISSUER = InterpolatedNodalCurve.of(
+      METADATA_ISSUER, new double[] {0.2, 9.0, 15.0}, new double[] {0.03, 0.5, 0.13}, INTERPOLATOR);
+  private static final DiscountFactors DSC_FACTORS_ISSUER = ZeroRateDiscountFactors.of(EUR, VALUATION, CURVE_ISSUER);
+  private static final LegalEntityGroup GROUP_ISSUER = LegalEntityGroup.of("GOVT1");
+  private static final LegalEntityDiscountingProvider PROVIDER = LegalEntityDiscountingProvider.builder()
+      .issuerCurves(ImmutableMap.<Pair<LegalEntityGroup, Currency>, DiscountFactors>of(
+          Pair.<LegalEntityGroup, Currency>of(GROUP_ISSUER, EUR), DSC_FACTORS_ISSUER))
+      .legalEntityMap(ImmutableMap.<StandardId, LegalEntityGroup>of(ISSUER_ID, GROUP_ISSUER))
+      .repoCurves(ImmutableMap.<Pair<BondGroup, Currency>, DiscountFactors>of(
+          Pair.<BondGroup, Currency>of(GROUP_REPO, EUR), DSC_FACTORS_REPO))
+      .bondMap(ImmutableMap.<StandardId, BondGroup>of(SECURITY_ID, GROUP_REPO))
+      .valuationDate(VALUATION)
+      .build();
+
+  // pricers
+  private static final DiscountingFixedCouponBondTradePricer TRADE_PRICER = DiscountingFixedCouponBondTradePricer.DEFAULT;
+  private static final DiscountingFixedCouponBondProductPricer PRODUCT_PRICER =
+      DiscountingFixedCouponBondProductPricer.DEFAULT;
+  private static final DiscountingPaymentPricer PRICER_NOMINAL = DiscountingPaymentPricer.DEFAULT;
+
+  private static final double Z_SPREAD = 0.035;
+  private static final int PERIOD_PER_YEAR = 4;
+  private static final double TOL = 1.0e-12;
+  private static final double EPS = 1.0e-6;
+
+  //-------------------------------------------------------------------------
+  public void test_presentValue() {
+    CurrencyAmount computedTrade = TRADE_PRICER.presentValue(TRADE, PROVIDER);
+    CurrencyAmount computedProduct = PRODUCT_PRICER.presentValue(PRODUCT, PROVIDER);
+    CurrencyAmount pvPayment = PRICER_NOMINAL.presentValue(UPFRONT_PAYMENT, DSC_FACTORS_REPO);
+    assertEquals(
+        computedTrade.getAmount(), computedProduct.multipliedBy(QUANTITY).plus(pvPayment).getAmount(), NOTIONAL * TOL);
+  }
+
```

Comment:
The present value tests should include the case where the settlement is one day before, on and one day after the coupon payment. The same test should also be done for an ex-coupon date for a UK govt bond (with ex-coupon period).